Classes/KIFUITestActor.h

```diff
@@ -477,6 +493,48 @@ static inline KIFDisplacement KIFDisplacementForSwipingInDirection(KIFSwipeDirec
 - (void)swipeAccessibilityElement:(UIAccessibilityElement *)element inView:(UIView *)viewToSwipe inDirection:(KIFSwipeDirection)direction;
 
 /*!
+ @abstract Pulls down on the view that enables the pull to refresh.
+ @discussion This will enact the pull to refresh by pulling down the distance of 1/2 the height of the view found by the accessibility label.
+ @param label The accessibility label of the view to swipe.
+ */
+- (void)pullToRefreshViewWithAccessibilityLabel:(NSString *)label;
+
+/*!
+ @abstract Pulls down on the view that enables the pull to refresh.
+ @discussion This will enact the pull to refresh by pulling down the distance of 1/2 the height of the view found by the accessibility label.
+ @param label The accessibility label of the view to swipe.
+ @param pullDownDuration The enum describing the approximate time for the pull down to travel the entire distance
+ */
+- (void)pullToRefreshViewWithAccessibilityLabel:(NSString *)label pullDownDuration:(KIFStepperPullToRefreshTiming) pullDownDuration;
+
+/*!
+ @abstract Pulls down on the view that enables the pull to refresh.
+ @discussion This will enact the pull to refresh by pulling down the distance of 1/2 the height of the view found by the accessibility label.
+ @param label The accessibility label of the view to swipe.
+ @param value The accessibility value of the view to swipe.
+ */
+- (void)pullToRefreshViewWithAccessibilityLabel:(NSString *)label value:(NSString *)value;
```

Comment:
I am able to remove 
- (void)pullToRefreshViewWithAccessibilityLabel:(NSString *)label value:(NSString *)value pullDownDuration:(KIFPullToRefreshTiming)pullDownDuration traits:(UIAccessibilityTraits)traits;


but below is needed for the identifier tests.

- (void)pullToRefreshAccessibilityElement:(UIAccessibilityElement *)element inView:(UIView *)viewToSwipe pullDownDuration:(KIFPullToRefreshTiming) pullDownDuration;






-----Original Message-----
From: phatmann <notifications@github.com>
To: kif-framework/KIF <KIF@noreply.github.com>
Cc: mikelupo <mikelupo@aol.com>
Sent: Fri, Sep 25, 2015 6:47 am
Subject: Re: [KIF] Add new pullToRefresh methods with associated label and identifier tests (#722)



In Classes/KIFUITestActor.h:
> +
> +/*!
> + @abstract Pulls down on the view that enables the pull to refresh.
> + @discussion This will enact the pull to refresh by pulling down the distance of 1/2 the height of the view found by the accessibility label.
> + @param label The accessibility label of the view to swipe.
> + @param pullDownDuration The enum describing the approximate time for the pull down to travel the entire distance
> + */
> +- (void)pullToRefreshViewWithAccessibilityLabel:(NSString *)label pullDownDuration:(KIFStepperPullToRefreshTiming) pullDownDuration;
> +
> +/*!
> + @abstract Pulls down on the view that enables the pull to refresh.
> + @discussion This will enact the pull to refresh by pulling down the distance of 1/2 the height of the view found by the accessibility label.
> + @param label The accessibility label of the view to swipe.
> + @param value The accessibility value of the view to swipe.
> + */
> +- (void)pullToRefreshViewWithAccessibilityLabel:(NSString *)label value:(NSString *)value;

Do we really need all three of the methods below in the public header file? I know that for methods like the "swipe" family we have done so, but for a high-level feature like pull to refresh I would think the two methods above would suffice.
—
Reply to this email directly or view it on GitHub.